src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    public static final int INITIAL_CAPACITY = 10;
+    private Object[] list;
+    private int size;
+
+    public ArrayList() {
+        this.list = new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
```

Comment:
You are coping arrays each time this method is called. Let's find another approach so that you're just adding a new value at the end of the array instead.